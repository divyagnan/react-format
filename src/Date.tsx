import * as React from "react";

export interface IDateProps {
  locale?: string;
  options?: {
    localeMatcher?: "lookup" | "best-fit";
    timeZone?: string;
    hour12?: boolean;
    formatMatcher?: "basic" | "best-fit";
    weekday?: "narrow" | "short" | "long";
    era?: "narrow" | "short" | "long";
    year?: "numeric" | "2-digit";
    month?: "numeric" | "2-digit" | "narrow" | "short" | "long";
    day?: "numeric" | "2-digit";
    hour?: "numeric" | "2-digit";
    minute?: "numeric" | "2-digit";
    second?: "numeric" | "2-digit";
    timeZoneName?: "short" | "long";
  };
  children: any;
  passthrough?: boolean;
}

const validDate = (potentialDate: any): boolean =>
  potentialDate instanceof Date || !!Date.parse(potentialDate);

const formatDate = (props: IDateProps): string =>
  Intl.DateTimeFormat(props.locale, props.options).format(
    Date.parse(props.children)
  );

export const DateFormatter: React.SFC<IDateProps> = (props): JSX.Element => (
  <span>
    {validDate(props.children) ? (
      formatDate(props)
    ) : (
      props.passthrough && props.children
    )}
  </span>
);

DateFormatter.defaultProps = {
  locale: "en-US",
  options: {},
  passthrough: false
};
