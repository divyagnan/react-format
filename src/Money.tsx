import * as React from "react";

export interface IMoneyProps {
  locale?: string;
  currency?: string;
  children: any;
}

export const Money: React.SFC<IMoneyProps> = (props): JSX.Element =>
  <span>
    {Intl.NumberFormat(props.locale, {
      style: "currency",
      currency: props.currency
    }).format(props.children)}
  </span>;

Money.defaultProps = {
  locale: "en-US",
  currency: "USD"
};
