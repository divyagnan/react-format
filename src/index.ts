export * from "./Money";
export { DateFormatter as Date, IDateProps } from "./Date";
