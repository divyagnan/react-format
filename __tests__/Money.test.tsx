import * as React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Money } from "../src/Money";

describe("Money Formatter", () => {
  it("does basic formatting - without config", () => {
    // Using the default props to have locale as "en-US" and currency as "USD"
    const wrapper = mount(<Money>3400</Money>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("accepts a locale prop", () => {
    const wrapper = mount(<Money locale="en-IN">3400</Money>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("accepts a currency prop", () => {
    const wrapper = mount(<Money currency="EUR">3400</Money>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("accepts a both a locale prop and a currency prop", () => {
    const wrapper = mount(<Money locale="ja-JP" currency="JPY">3400</Money>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
