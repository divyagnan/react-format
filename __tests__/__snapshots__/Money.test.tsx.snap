// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money Formatter accepts a both a locale prop and a currency prop 1`] = `
<Component
  currency="EUR"
  locale="de-DE"
>
  <span>
    3.400,00 €
  </span>
</Component>
`;

exports[`Money Formatter accepts a currency prop 1`] = `
<Component
  currency="EUR"
  locale="en-US"
>
  <span>
    €3,400.00
  </span>
</Component>
`;

exports[`Money Formatter accepts a locale prop 1`] = `
<Component
  currency="USD"
  locale="de-DE"
>
  <span>
    3.400,00 $
  </span>
</Component>
`;

exports[`Money Formatter does basic formatting - without config 1`] = `
<Component
  currency="USD"
  locale="en-US"
>
  <span>
    $3,400.00
  </span>
</Component>
`;
